internal/collector/instance.go

```diff
@@ -136,18 +150,22 @@ func AddToPod(
 			}},
 		}
 		container.VolumeMounts = append(container.VolumeMounts, logrotateConfigVolumeMount)
-		outPod.Volumes = append(outPod.Volumes, logrotateConfigVolume)
+		template.Spec.Volumes = append(template.Spec.Volumes, logrotateConfigVolume)
 	}
 
 	if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
 		container.Ports = []corev1.ContainerPort{{
-			ContainerPort: int32(8889),
+			ContainerPort: int32(PrometheusPort),
 			Name:          "otel-metrics",
 			Protocol:      corev1.ProtocolTCP,
 		}}
 	}
 
-	outPod.Containers = append(outPod.Containers, container)
+	template.Spec.Containers = append(template.Spec.Containers, container)
+
+	// add the proper label to support Pod discovery by Prometheus
+	initialize.Labels(template)
+	template.Labels[naming.LabelCollectorDiscovery] = "true"
```

Comment:
Should all the labeling and the metric-specific file work be done inside the `if feature.Enabled(ctx, feature.OpenTelemetryMetrics)` block?

Reply:
As indicated in the conversation below, I think the label should go on all Pods that have the collector in place, but I agree that the projecting of the custom queries into the Pod should only happen if metrics is enabled.